pages/c/[communityName].js

```diff
@@ -0,0 +1,48 @@
+import Constant from "utils/constant";
+import RedirectUtils from "utils/redirect"
+import UserAgentUtils from "utils/userAgent"
+import parser from 'ua-parser-js';
+import { useEffect } from "react"
+import { useRouter } from "next/router"
+
+export async function getServerSideProps(context) {
+    const { communityName } = context.query
+    let isDynamicLink = communityName?.indexOf('+') >= 0
+    let userAgent = parser(context?.req?.headers['user-agent'])
+    
+    if (UserAgentUtils.isMobile(userAgent)) {
+        let redirect = await RedirectUtils.redirectCommunityForMobileDevice(userAgent, communityName)
+        if (redirect && !isDynamicLink) return redirect
+    }
+
+    if(UserAgentUtils.isIos(userAgent)) {
+        return {
+            redirect: {
+                destination: Constant.Link.appstore,
+            }
+        }
+    }
+
+    if(UserAgentUtils.isAndroid(userAgent)) {
+        return {
+            redirect: {
+                destination: Constant.Link.playstore,
+            }
+        }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, this section redirects Android users to the Play Store without checking whether the app is already installed. Consider adding a check for this scenario as well.


<!-- This is an auto-generated comment by OSS CodeRabbit -->